docs/adr/0019-automatic-process-creation.md

```diff
@@ -0,0 +1,46 @@
+# 16. Automatic Process Creation
+
+Date: 2021-02-18
+
+## Status
+
+Accepted
+
+## Context
+
+After reviewing and reworking the aggregate API in
+[ADR-16](0016-automatic-aggregate-creation.md) and
+[ADR-17](0017-recreate-aggregate-after-destruction.md) we have conducted a
+similar review of the process API in an effort to both simplify the API and
+improve consistency between the aggregate and process APIs.
+
+## Decision
+
+- Remove `ProcessEventScope.Begin()`.
+- Remove `Process[Event|Timeout]Scope.HasBegun()`.
+- Remove `Process[Event|Timeout]Scope.Root()`.
+- Pass the process root directly to `ProcessMessageHandler.Handle[Event|Timeout]()`.
+- Allow `Process[EventTimeout]Scope.End()` to be called at any time.
+- Allow `Process[EventTimeout]Scope.ExecuteCommand()` and `ScheduleTimeout()` to
+  be called at any time. Doing so should "negate" any prior call to `End()` as
+  though it were never called.
+- Remove `StatelessProcessRoot` and `StatelessProcessBehavior`. With the
+  notion of "existence" being removed from the public API a stateless process
+  becomes nonsensical.
+- Routing a command message to a process instance causes that instance to begin.
+
+## Consequences
+
+Largely, this should simplify implementations of both `ProcessMessageHandler` by
+application developers and `Process[Event|Timeout]Scope` by engine developers.
+
+With the removal of `StatelessProcessRoot`, some process implementations will
+require an additional type declaration for their root value.
+
+The removal of `Begin()` does mean that engines longer get an explicit request
```

Comment:
```suggestion
The removal of `Begin()` does mean that engines get an explicit request longer
```
May be this adverb should be relocated?

Reply:
Oh I totally missed that, good catch. I think it's just missing a "no":
```suggestion
The removal of `Begin()` does mean that engines no longer get an explicit request
```

Reply:
Yeah I guess it make more sense this way 👍

Reply:
Done!